templates/company-board/contracts/CompanyBoardTemplate.sol

```diff
@@ -48,42 +39,90 @@ contract CompanyBoardTemplate is BaseTemplate {
         _ensureMiniMeFactoryIsValid(_miniMeFactory);
     }
 
-    function prepareInstance() public {
+    function prepareInstance(string _shareTokenName, string _shareTokenSymbol, uint64 _financePeriod) public {
         (Kernel dao,) = _createDAO();
         MiniMeToken boardToken = _createToken(BOARD_TOKEN_NAME, BOARD_TOKEN_SYMBOL, BOARD_TOKEN_DECIMALS);
-        MiniMeToken shareToken = _createToken(SHARE_TOKEN_NAME, SHARE_TOKEN_SYMBOL, SHARE_TOKEN_DECIMALS);
-        _storeCache(dao, boardToken, shareToken, msg.sender);
+        MiniMeToken shareToken = _createToken(_shareTokenName, _shareTokenSymbol, SHARE_TOKEN_DECIMALS);
+        (Agent agent, Finance finance) = _setupCommon(dao, _financePeriod);
+        _storeCache(dao, boardToken, shareToken, agent, finance, msg.sender);
+    }
+
+    function setupInstance(
+        string _id, 
+        address[] _boardMembers, 
+        address[] _shareHolders, 
+        uint256[] _shareStakes,
+        uint64 _boardVoteDuration,
+        uint64 _shareVoteDuration,
+        uint64 _boardSupportRequired,
+        uint64 _shareSupportRequired,
+        uint64 _boardMinAcceptanceQuorum,
+        uint64 _shareMinAcceptanceQuorum
```

Comment:
I realized that we could use 2 lists of 3 `uint64` elements each to specify the voting app's parameters. This would allow us to reduce the number of declared variables and maybe we can move the finance and agent apps setup to this step, which will reduce gas costs since we won't need to cache them

Reply:
Nize!
Applied. It does make the code much cleaner, but it doesn't save gas :(

Reply:
Probably not yet for this transaction, but the total should have been reduced since we are not storing the finance and agent apps any more